### ISSUE TITLE
Unraid m3u playlist never loads

### ISSUE THREAD

**Issue by Wellsj27** [Threadfin/Threadfin#398]
**Describe the bug**
When first starting the app i add my playliist and it never loads the playlist it shows in the logs it is trying but never gets passed it.

**Server (please complete the following information):**
- OS UNraid 6.12.10
- intel
- Threadfin Version and Build  1.2 build 9

**To Reproduce**
1. load app and add playlist

**Expected behavior**
loads play list and lets me continue. I have waited over 20 minutes multiple times to not see it load

**Screenshots**
<img width="652" alt="Screenshot 2024-10-11 at 9 34 58 PM" src="https://github.com/user-attachments/assets/376d916b-52ee-46df-8b6d-677b8b463178">

**Desktop (please complete the following information):**
 - OS: unraid
 - Browser firefox

**Additional context**

**Comment by ovargasp** [comment]
If you're using m3u file as a playlist you may want to take a look at this https://github.com/Threadfin/Threadfin/issues/376#issuecomment-2351821824  to see if it helps...  In the screenshot I noticed the PMS configured and that may be the issue. I hope it helps

**Comment by bgraves49** [comment]
I have the same issue, went back to 1.2.1 and it works fine.  Its a pretty large m3u list but still works in xteve and older Threadfin.

**Comment by ovargasp** [comment]
I'm using a 430+ list of channels in 1.2.9 with no problems so far. Threadfin puts a 480 channels limit to the sum of all playlists channels due a Plex software limitation. That will cause all channels to be disabled but visible in Filter tab.
I proposed some code yesterday to change that limit if you're using Jellyfin/Emby or some player that doesn't have Plex software restriction.

**Comment by bgraves49** [comment]
> I'm using a 430+ list of channels in 1.2.9 with no problems so far. Threadfin puts a 480 channels limit to the sum of all playlists channels due a Plex software limitation. That will cause all channels to be disabled but visible in Filter tab. I proposed some code yesterday to change that limit if you're using Jellyfin/Emby or some player that doesn't have Plex software restriction.

My playlist has 258,314 streams in it.  Only 45 are filtered and mapped.    Using older version until it's sorted (1.2.1).    I switched from xteve which was working fine.  Granite it takes 2-3 minutes to load (expected), it does load.   On the newer and newest version, it stalls and never goes anywhere.    Listing the exact screenshot as above from someone else.  "Check File"

**Comment by ovargasp** [comment]
That's some use case :scream:, I'm not familiar with all the code, I just started using Threadfin a few weeks ago, and I know the code that I've read that there is one limitation for Plex, after all the project is called **M3U Proxy for Plex DVR** and as you said if they added a sorting algorith for the list, it may also have a setting/limit for the quantity of channels/lines it can sort. But it's good to know that older versions aren't limited by this parameter, as I told you I posted a proposal to change the code and allow more channels but I came out very short to your use case...

**Comment by Fyb3roptik** [comment]
Your source needs to be XEPG not PMS

**Comment by bgraves49** [comment]
My personal configuration is XPEG with a valid URL (which loads) and it still cannot be upgraded to something newer without stalling when adding the list 😞

**Comment by ovargasp** [comment]
@bgraves49 I'm not sure if Threadfin's newer versions will be able to handle your specific playlist, but I've been experimenting and managed to load and map 11,842 channels. This is the largest playlist I've been able to access, and with the recent changes I've proposed (and some more I'll be pushing soon), Threadfin was able to handle it. 

![image](https://github.com/user-attachments/assets/61b6cba6-cbe0-4b06-9d74-14140b6da0b1)

this is how the player that I'm using shows the list.

![image](https://github.com/user-attachments/assets/4a4e2011-6914-4d0b-a95c-07b91adf33a6)


@Fyb3roptik is making a great job with Threadfin, but it's complicated to cover all use cases, so I'm not sure your personal configuration is on the development roadmap.
